underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -261,7 +307,90 @@ public boolean mkdirs(String path, boolean createParent) throws IOException {
 
   @Override
   public boolean mkdirs(String path, MkdirsOptions options) throws IOException {
-    return true;
+    if (path == null) {
+      LOG.error("Attempting to create directory with a null path");
+      return false;
+    }
+    if (isDirectory(path)) {
+      return true;
+    }
+    if (isFile(path)) {
+      LOG.error("Cannot create directory {} because it is already a file.", path);
+      return false;
+    }
+
+    if (!parentExists(path)) {
+      if (!options.getCreateParent()) {
+        LOG.error("Cannot create directory {} because parent does not exist", path);
+        return false;
+      }
+      final String parentPath = getParentPath(path);
+      // Recursively make the parent folders
+      if (!mkdirs(parentPath, true)) {
```

Comment:
This recursive call is quite expensive, right? We will be calling `isDirectory`, `isFile`, `parentExists`, for every parent component that doesn't exist. That will be at least 4 RPCs per parent component?

Maybe add a TODO to get better performance from this?